crates/ruff_linter/src/rules/ruff/rules/unnecessary_nested_literal.rs

```diff
@@ -0,0 +1,129 @@
+use ruff_diagnostics::{Applicability, Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{Expr, ExprContext, ExprSubscript, ExprTuple};
+use ruff_python_semantic::analyze::typing::traverse_literal;
+use ruff_text_size::{Ranged, TextRange};
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for unnecessary nested `Literal`.
+///
+/// ## Why is this bad?
```

Comment:
The direct answer to the question of "Why is this bad?" is "it's less readable than the alternative". I think something to that effect should be the first sentence of this section, as it makes it clear that this rule is about readability and style rather than correctness.